modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -979,6 +1018,94 @@ private CPDefinition _updateNestedResources(
+					GetterUtil.getString(
+						productConfiguration.getExternalReferenceCode(),
+						masterCPConfigurationEntry.getExternalReferenceCode()),
+					masterCPConfigurationEntry.getCPConfigurationEntryId(),
+					GetterUtil.getLong(
+						productTaxConfiguration.getId(),
+						masterCPConfigurationEntry.getCPTaxCategoryId()),
+					ProductConfigurationUtil.getAllowedOrderQuantities(
+						productConfiguration.getAllowedOrderQuantities(),
+						masterCPConfigurationEntry.getAllowedOrderQuantities()),
+					GetterUtil.getBoolean(
+						productConfiguration.getAllowBackOrder(),
+						masterCPConfigurationEntry.isBackOrders()),
+					GetterUtil.getLong(
+						productConfiguration.getAvailabilityEstimateId(),
+						masterCPConfigurationEntry.
+							getCommerceAvailabilityEstimateId()),
+					GetterUtil.getString(
+						productConfiguration.getInventoryEngine(),
+						masterCPConfigurationEntry.
+							getCPDefinitionInventoryEngine()),
+					GetterUtil.getDouble(
+						productShippingConfiguration.getDepth(),
+						masterCPConfigurationEntry.getDepth()),
+					GetterUtil.getBoolean(
+						productConfiguration.getDisplayAvailability(),
+						masterCPConfigurationEntry.isDisplayAvailability()),
+					GetterUtil.getBoolean(
+						productConfiguration.getDisplayStockQuantity(),
+						masterCPConfigurationEntry.isDisplayStockQuantity()),
+					GetterUtil.getBoolean(
+						productShippingConfiguration.getFreeShipping(),
+						masterCPConfigurationEntry.isFreeShipping()),
+					GetterUtil.getDouble(
+						productShippingConfiguration.getHeight(),
+						masterCPConfigurationEntry.getHeight()),
+					GetterUtil.getString(
+						productConfiguration.getLowStockAction(),
+						masterCPConfigurationEntry.getLowStockActivity()),
+					BigDecimalUtil.get(
+						productConfiguration.getMaxOrderQuantity(),
+						masterCPConfigurationEntry.getMaxOrderQuantity()),
+					BigDecimalUtil.get(
+						productConfiguration.getMinOrderQuantity(),
+						masterCPConfigurationEntry.getMinOrderQuantity()),
+					BigDecimalUtil.get(
+						productConfiguration.getMinStockQuantity(),
+						masterCPConfigurationEntry.getMinStockQuantity()),
+					BigDecimalUtil.get(
+						productConfiguration.getMultipleOrderQuantity(),
+						masterCPConfigurationEntry.getMultipleOrderQuantity()),
+					GetterUtil.getBoolean(
+						productConfiguration.getPurchasable(),
+						masterCPConfigurationEntry.isPurchasable()),
+					GetterUtil.getBoolean(
+						productShippingConfiguration.getShippable(),
+						masterCPConfigurationEntry.isShippable()),
+					GetterUtil.getDouble(
+						productShippingConfiguration.getShippingExtraPrice(),
+						masterCPConfigurationEntry.getShippingExtraPrice()),
+					GetterUtil.getBoolean(
+						productShippingConfiguration.getShippingSeparately(),
+						masterCPConfigurationEntry.isShipSeparately()),
+					GetterUtil.getBoolean(
+						!_isTaxable(productTaxConfiguration),
+						masterCPConfigurationEntry.isTaxExempt()),
```

Comment:
why here we need to have a private method? Also because in this scenario the GetterUtil.getBoolean here never reaches the "second parameter" because _isTaxable has always a value